### DISCUSSION TITLE
ebyte E32-DTU settings/ correlation

### DISCUSSION THREAD

**Discussion by Ruboninov** [sandeepmistry/arduino-LoRa#606] [Q&A] [answered]
Hello,
I would like a LoRa modem talk to a TTGO device using the library here.
However I cannot figure out on how to correlate the settings.

The arduino LoRa library uses "Signal Bandwidth" setting
Supported values are 7.8E3, 10.4E3, 15.6E3, 20.8E3, 31.25E3, 41.7E3, 62.5E3, 125E3, 250E3, and 500E3.

However, the ebyte E32 uses an "AirRate" setting
![grafik](https://user-images.githubusercontent.com/77238711/201541833-828261c1-5edc-4eaf-ae9e-a8753e8662f0.png)

Values don't seem to correspondent.
Any idea how to correlate the settings on both devices?

**Comment by Kongduino** [comment, marked as answer]
That's because the ebyte -T- models (UART driven) don't do LoRa but FSK. AirRate is not a LoRa setting. Unless the SPI pins are exposed, and you can connect the ebyte module to an Arduino or else, you are stuck in their ecosystem.

**Reply by Ruboninov** [reply]
Thank You for this information!
So, You mean despite their manual stating "LoRa spread spectrum", they don't do LoRa? (I got model E32-DTU 433L30)
![grafik](https://user-images.githubusercontent.com/77238711/202161909-9579544f-5b8a-469d-a143-4a2bc43223cc.png)
...or do they just put another "Ebyte"-Layer on top of LoRa, jailing You to their ecosystem by using LoRa, but as a black box, therefor LoRa parameters not to be accessible to the consumer?

**Comment by Kongduino** [comment]
LoRa chips can do more than LoRa. And ebyte does just that with their -T- model. I struggled with that part many years ago, even enlisted the help of a Semtech engineer – whose final comment was whatever TF they're doing that's not LoRa. Turns out `AirRate` (aka BitRate) is a typical setting for FSK/OOK. I'm pretty sure if I could connect SPI to one of these, in the case of an SX127x anyway, reading register RegOpMode (0x01) would return 0 for bit 7.

Likewise, reading registers RegBitrateMsb (0x02) and RegBitrateLsb (0x03) would give me 4.8 kb/s I bet...

![sx127x settings](https://user-images.githubusercontent.com/35363882/202204211-817416fc-1cc8-4106-9a16-977aef149d72.png)

It's a cheap and easy way to lock you in their ecosystem. The -T- models are managed via a small MCU, usually an atmega or something similar, locking you in. IF the SPI pins are available, it's child's play to get rid of the limitations. If not, oops.
